tests/test_trainer.py

```diff
@@ -1058,6 +1068,28 @@ def test_training_iterable_dataset(self):
         self.assertIsInstance(loader, torch.utils.data.DataLoader)
         self.assertIsInstance(loader.sampler, torch.utils.data.dataloader._InfiniteConstantSampler)
 
+    def test_training_finite_iterable_dataset(self):
+        config = RegressionModelConfig()
+        model = RegressionPreTrainedModel(config)
+
+        batch_size = 1
+        gradient_accumulation_steps = 1
+        num_samples = 10
+
+        available_steps = num_samples // (batch_size * gradient_accumulation_steps)
```

Comment:
We're not really using `gradient_accumulation_steps` here so let's remove.
```suggestion
        batch_size = 1
        num_samples = 10

        available_steps = num_samples // batch_size
```

Reply:
Right, I falsely assumed there was another default value for it than 1. I removed it.